src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTestData.cs

```diff
@@ -0,0 +1,190 @@
+                    break;
+                case "Sm":
+                    generalCategory = UnicodeCategory.MathSymbol;
+                    break;
+                case "Lm":
+                    generalCategory = UnicodeCategory.ModifierLetter;
+                    break;
+                case "Sk":
+                    generalCategory = UnicodeCategory.ModifierSymbol;
+                    break;
+                case "Mn":
+                    generalCategory = UnicodeCategory.NonSpacingMark;
+                    break;
+                case "Ps":
+                    generalCategory = UnicodeCategory.OpenPunctuation;
+                    break;
+                case "Lo":
+                    generalCategory = UnicodeCategory.OtherLetter;
+                    break;
+                case "Cn":
+                    generalCategory = UnicodeCategory.OtherNotAssigned;
+                    break;
+                case "No":
+                    generalCategory = UnicodeCategory.OtherNumber;
+                    break;
+                case "Po":
+                    generalCategory = UnicodeCategory.OtherPunctuation;
+                    break;
+                case "So":
+                    generalCategory = UnicodeCategory.OtherSymbol;
+                    break;
+                case "Zp":
+                    generalCategory = UnicodeCategory.ParagraphSeparator;
+                    break;
+                case "Co":
+                    generalCategory = UnicodeCategory.PrivateUse;
+                    break;
+                case "Zs":
+                    generalCategory = UnicodeCategory.SpaceSeparator;
+                    break;
+                case "Mc":
+                    generalCategory = UnicodeCategory.SpacingCombiningMark;
+                    break;
+                case "Cs":
+                    generalCategory = UnicodeCategory.Surrogate;
+                    break;
+                case "Lt":
+                    generalCategory = UnicodeCategory.TitlecaseLetter;
+                    break;
+                case "Lu":
+                    generalCategory = UnicodeCategory.UppercaseLetter;
+                    break;
+                default:
+                    throw new InvalidOperationException("No such UnicodeCharCategory. Check the test data.");
+            }
+
+            return new CharUnicodeInfoTestCase()
+            {
+                Utf32CodeValue = codeValueRepresentation,
+                GeneralCategory = generalCategory,
+                NumericValue = numericValue
+            };
+        }
+
+        private static double ParseNumericValueString(string numericValueString)
+        {
+            if (numericValueString.Length == 0)
+            {
+                // Parsing empty string (no numeric value)
+                return -1;
+            }
+
+            int fractionDelimeterIndex = numericValueString.IndexOf("/", StringComparison.Ordinal);
```

Comment:
@hughbe, that's if you're searching for a string... searching for a char is always ordinal.